mod_ood_proxy/lib/ood/user_map.lua

```diff
@@ -41,7 +41,14 @@ function map(r, user_map_match, user_map_cmd, remote_user)
   local sys_user = ""
   -- match string
   if user_map_match ~= nil then
-    sys_user = string.match(remote_user, user_map_match)
+    -- if match string ends in :lower, lowercase the username
+    local pat = string.match(user_map_match, "^(.*):lower$")
```

Comment:
I'm going to have to think about this for a bit. Not in love with using a regex to see if it matches another regex. Though `string.find` does the same thing really...

Reply:
I hear you. Just less in love with calling shell scripts from apache with the power to change users, just to lower case a username. I figure this is 99% of the use cases for the map command, calling `echo ${1,,}` in a bash script sitting in /opt/ood, hopefully with the right permissions...

I suppose you could add a flag instead. More places to pass around, but perhaps cleaner.

Or, honestly, you could just _always_ `lower()` the username. People who, for reasons I don't want to contemplate, need upper case characters in their usernames can always resort to map_command. It wouldn't be backwards compatible, but hopefully not affect anyone really.

Reply:
I mean, it's not like the current implementation is broken. This is just more convenient, and perhaps a tiny bit more secure. I don't see that a more complex implementation will gain anything to justify the extra lines of code.

I don't mind if you decide to close this. I'd close the ticket, too though. No point keeping the idea around forever, you'll just spend time on it every time you try to clean up the backlog.

Reply:
I'm fairly sure we can move forward with it, sometimes I just need a second to walk away and be sure.

Regarding documentation - yes please create a PR for documentation when this gets merged!

Reply:
I hear you. Will do docs. Do I need to sign of on a contributor thingy somewhere, too, or you good to accept as is?

Reply:
It's fine.